editors/code/build.mjs

```diff
@@ -0,0 +1,94 @@
+import * as path from "node:path";
+import { parseArgs } from "node:util";
+import * as esbuild from "esbuild";
+
+function parseCliOptions() {
+    const { values } = parseArgs({
+        options: {
+            minify: {
+                type: "boolean",
+                default: false,
+            },
+            sourcemap: {
+                type: "boolean",
+                default: false,
+            },
+            watch: {
+                type: "boolean",
+                default: false,
+            },
+        },
+        strict: true,
+    });
+    return {
+        shouldMinify: !!values.minify,
+        shouldEmitSourceMap: !!values.sourcemap,
+        isWatchMode: !!values.watch,
+    };
+}
+
+const { shouldMinify, shouldEmitSourceMap, isWatchMode } = parseCliOptions();
+
+const OUT_DIR = "./out";
+const OUT_WEBVIEW_DIR = path.resolve(OUT_DIR, "webview");
+
+/** @type {esbuild.BuildOptions} */
+const BASE_OPTIONS = {
+    minify: shouldMinify,
+    sourcemap: shouldEmitSourceMap ? "external" : false,
+    bundle: true,
+};
+
+function createBuildOption(entryPoints) {
+    /** @type {esbuild.BuildOptions} */
+    const options = {
+        ...BASE_OPTIONS,
+        entryPoints,
+        external: ["vscode"],
+        format: "cjs",
+        platform: "node",
+        target: "node16",
+        outdir: OUT_DIR,
+    };
+    return options;
+}
+
+function createBuildOptionForWebView(entryPoints) {
+    /** @type {esbuild.BuildOptions} */
+    const options = {
+        ...BASE_OPTIONS,
+        entryPoints,
+        format: "esm",
+        platform: "browser",
+        // VSCode v1.78 (Electron 22) uses Chromium 108.
```

Comment:
We're back on 1.75 since https://github.com/rust-lang/rust-analyzer/pull/15333.

Reply:
- [v1.78 updates to Electron 22](https://code.visualstudio.com/updates/v1_78).
- Before v1.78, [vscode uses Electron 19](https://code.visualstudio.com/updates/v1_71#_electron-19-update) (Chromium 102.0.5005.167 and Node.js 16.17.1).

I'll update to:

```diff
- // VSCode v1.78 (Electron 22) uses Chromium 108.
+ // VSCode v1.75 (Electron 19) uses Chromium 102.
```